packages/mobile/src/identity/privacy.ts

```diff
@@ -0,0 +1,138 @@
+      Logger.debug(`${TAG}@fetchPrivatePhoneHash`, 'Salt was cached')
+      const phoneHash = getPhoneHash(e164Number, cachedSalt)
+      return { e164Number, phoneHash, salt: cachedSalt }
+    }
+
+    Logger.debug(`${TAG}@fetchPrivatePhoneHash`, 'Salt was not cached, fetching')
+    const details: PhoneNumberHashDetails = yield call(getPhoneHashPrivate, e164Number)
+    yield put(updateE164PhoneNumberSalts({ [e164Number]: details.salt }))
+    return details
+  } catch (error) {
+    if (error.message === ErrorMessages.SALT_QUOTA_EXCEEDED) {
+      Logger.error(
+        `${TAG}@fetchPrivatePhoneHash`,
+        'Salt quota exceeded, navigating to quota purchase screen'
+      )
+      // TODO nav to quota purchase screen
+    } else {
+      throw new Error(ErrorMessages.SALT_FETCH_FAILURE)
+    }
+  }
+}
+
+// Unlike the getPhoneHash in utils, this leverage the phone number
+// privacy service to compute a secure, unique salt for the phone number
+// and then appends it before hashing.
+async function getPhoneHashPrivate(e164Number: string): Promise<PhoneNumberHashDetails> {
+  const salt = await getPhoneNumberSalt(e164Number)
+  const phoneHash = getPhoneHash(e164Number, salt)
+  return {
+    e164Number,
+    phoneHash,
+    salt,
+  }
+}
+
+async function getPhoneNumberSalt(e164Number: string) {
+  Logger.debug(`${TAG}@getPhoneNumberSalt`, 'Getting phone number salt')
+
+  if (!isE164Number(e164Number)) {
+    throw new Error(ErrorMessages.INVALID_PHONE_NUMBER)
+  }
+
+  // TODO remove when iOS works
+  if (Platform.OS === 'ios') {
+    return 'fakeSalt'
+  }
+
+  Logger.debug(`${TAG}@getPhoneNumberSalt`, 'Retrieving blinded message')
+  const base64BlindedMessage = await BlindThresholdBls.blindMessage(e164Number)
+  const base64BlindSig = await postToSignMessage(base64BlindedMessage)
+  Logger.debug(`${TAG}@getPhoneNumberSalt`, 'Retrieving unblinded signature')
+  const base64UnblindedSig = await BlindThresholdBls.unblindMessage(base64BlindSig)
+  Logger.debug(`${TAG}@getPhoneNumberSalt`, 'Converting sig to salt')
+  return getSaltFromThresholdSignature(base64UnblindedSig)
+}
+
+interface SignMessageResponse {
+  success: boolean
+  // TODO change this in service
+  salt: string
+}
+
+// Send the blinded message off to the phone number privacy service and
+// get back the theshold signed blinded message
+async function postToSignMessage(base64BlindedMessage: string) {
+  Logger.debug(`${TAG}@postToSignMessage`, 'Retrieving blinded message')
+  const res = await fetch(PHONE_NUM_PRIVACY_SERVICE + SIGN_MESSAGE_ENDPOINT, {
+    method: 'POST',
+    headers: {
+      Accept: 'application/json',
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      blindPhoneNumber: base64BlindedMessage,
+    }),
+  })
+
+  if (!res.ok) {
+    handleSignMessageFailure(res)
+  }
+
+  Logger.debug(`${TAG}@postToSignMessage`, 'Response ok. Parsing.')
+  const signResponse = (await res.json()) as SignMessageResponse
```

Comment:
Do we want to `try ... catch` here?

Reply:
I'd rather let it propagate and have the try/catch in `fetchPhoneHashPrivate` handle it